Tests/SentryProfilerTests/SentryAppLaunchProfilingTests.swift

```diff
@@ -14,25 +14,98 @@ final class SentryAppLaunchProfilingSwiftTests: XCTestCase {
+
+        // -- Act --
+        let options = Options()
+        options.enableAppLaunchProfiling = true
+        options.profilesSampleRate = expectedProfilesSampleRate
+        options.tracesSampleRate = expectedTracesSampleRate
+
+        // Smoke test that the file doesn't exist yet
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        sentry_sdkInitProfilerTasks(options, TestHub(client: nil, andScope: nil))
+
+        // -- Assert --
+        XCTAssert(appLaunchProfileConfigFileExists())
+        let dict = try XCTUnwrap(sentry_appLaunchProfileConfiguration())
+        XCTAssertEqual(dict[kSentryLaunchProfileConfigKeyTracesSampleRate], expectedTracesSampleRate)
+        XCTAssertEqual(dict[kSentryLaunchProfileConfigKeyTracesSampleRand], expectedTracesSampleRand)
+        XCTAssertEqual(dict[kSentryLaunchProfileConfigKeyProfilesSampleRate], expectedProfilesSampleRate)
+        XCTAssertEqual(dict[kSentryLaunchProfileConfigKeyProfilesSampleRand], expectedProfilesSampleRand)
+    }
+
+    // test that after configuring for a launch profile, a subsequent
+    // configuration with insufficient sample rates removes the configuration
+    // file
+    func testLaunchTraceProfileConfigurationRemoval() {
+        let options = Options()
+        options.enableAppLaunchProfiling = true
+        options.profilesSampleRate = 0.567
+        options.tracesSampleRate = 0.789
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        sentry_sdkInitProfilerTasks(options, TestHub(client: nil, andScope: nil))
+        XCTAssert(appLaunchProfileConfigFileExists())
+        options.profilesSampleRate = 0.1 // less than the fixture's "random" value of 0.5
+        sentry_sdkInitProfilerTasks(options, TestHub(client: nil, andScope: nil))
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        // ensure we get another config written, to test removal again
+        options.profilesSampleRate = 0.567
+        sentry_sdkInitProfilerTasks(options, TestHub(client: nil, andScope: nil))
+        XCTAssert(appLaunchProfileConfigFileExists())
+        options.tracesSampleRate = 0
+        sentry_sdkInitProfilerTasks(options, TestHub(client: nil, andScope: nil))
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+    }
+
+    func testTraceProfilerStartsWhenBothSampleRatesAreSetAboveZero() {
+        let options = Options()
+        options.enableAppLaunchProfiling = true
+        options.profilesSampleRate = 0.567
+        options.tracesSampleRate = 0.789
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        sentry_sdkInitProfilerTasks(options, TestHub(client: nil, andScope: nil))
+        XCTAssertTrue(appLaunchProfileConfigFileExists())
+        _sentry_nondeduplicated_startLaunchProfile()
+        XCTAssert(SentryTraceProfiler.isCurrentlyProfiling())
+    }
+}
+
+// MARK: transaction based profiling iOS-only
+#if !os(macOS)
```

Comment:
`m`: instead of removing the entire tests it would be great to keep the tests and use `XCTSkip` in each test instead

Reply:
the API inside the test bodies doesn't compile on other platforms, we'd have to also surround the test body by `#if available`, i don't think there's value in doing this